pkg/ccl/changefeedccl/compression.go

```diff
@@ -17,24 +19,110 @@ import (
 	"github.com/klauspost/pgzip"
 )
 
-var useFastGzip = settings.RegisterBoolSetting(
-	settings.ApplicationLevel,
-	"changefeed.fast_gzip.enabled",
-	"use fast gzip implementation",
-	metamorphic.ConstantWithTestBool(
-		"changefeed.fast_gzip.enabled", true,
-	),
-	settings.WithPublic)
+var (
+	useFastGzip = settings.RegisterBoolSetting(
+		settings.ApplicationLevel,
+		"changefeed.fast_gzip.enabled",
+		"use fast gzip implementation",
+		metamorphic.ConstantWithTestBool(
+			"changefeed.fast_gzip.enabled", true,
+		),
+		settings.WithPublic)
+	gzipEncoderPool     = sync.Pool{}
+	fastGzipEncoderPool = sync.Pool{}
+	zstdEncoderPool     = sync.Pool{}
+	gzipDecoderPool     = sync.Pool{}
+	zstdDecoderPool     = sync.Pool{}
+)
 
 type compressionAlgo string
 
-const sinkCompressionGzip compressionAlgo = "gzip"
-const sinkCompressionZstd compressionAlgo = "zstd"
+const (
+	sinkCompressionGzip compressionAlgo = "gzip"
+	sinkCompressionZstd compressionAlgo = "zstd"
+)
 
 func (a compressionAlgo) enabled() bool {
 	return a != ""
 }
 
+type flusher interface {
+	Flush() error
+}
+
```

Comment:
I dont think we need to make an interface for this. Can you just include Flush() in the encoder interface? ditto for resettableEncoder, resettableDecoder

Reply:
Agree. Removed